client/rpc/trace/src/lib.rs

```diff
@@ -0,0 +1,417 @@
+			sp_utils::mpsc::tracing_unbounded("trace-filter-cache-requester");
+
+		let fut = async move {
+			let mut expiration_futures = FuturesUnordered::new();
+			// Substrate block hash => Block cache
+			let mut cached_blocks = BTreeMap::<H256, CacheBlock>::new();
+
+			tracing::info!("Begining Trace Filter Cache Task ...");
+
+			loop {
+				select! {
+					req = rx.next() => {
+						if let Some((req, response_tx)) = req {
+							let span = tracing::debug_span!("received request", request = ?req);
+							let _guard = span.enter();
+
+							tracing::trace!("Begining handling request");
+
+							let mut touched_blocks = vec![];
+							let res = Self::handle_request(
+								&client,
+								&backend,
+								&mut cached_blocks,
+								req,
+								&mut touched_blocks
+							);
+
+							expiration_futures.push(async move {
+								delay_for(Duration::from_secs(60)).await;
+								touched_blocks
+							});
+
+							tracing::trace!(?res, "Work done, sending response ...");
+
+							// Send response.
+							let _ = response_tx.send(res);
+						} else {
+							// All Senders are dropped, stopping the service.
+							break;
+						}
+					},
+					blocks_to_check = expiration_futures.next() => {
+						if let Some(blocks_to_check) = blocks_to_check {
+							tracing::trace!(?blocks_to_check, "Waking up for potential cache cleaning ...");
+
+							let now = Instant::now();
+
+							let mut blocks_to_remove = vec![];
+
+							for block in blocks_to_check {
+								if let Some(cache) = cached_blocks.get(&block) {
+									if cache.expiration <= now {
+										tracing::trace!(%block, "Removing expired block");
+
+										blocks_to_remove.push(block);
+									}
+								}
+							}
+
+							for block in blocks_to_remove {
+								let _ = cached_blocks.remove(&block);
+							}
+						}
+					},
+				}
+			}
+		}
+		.instrument(tracing::debug_span!("trace_filter_cache"));
+
+		(fut, tx)
+	}
+
+	fn block_id(client: &C, id: Option<RequestBlockId>) -> Result<u32> {
+		match id {
+			Some(RequestBlockId::Number(n)) => Ok(n),
+			None | Some(RequestBlockId::Tag(RequestBlockTag::Latest)) => {
+				Ok(client.info().best_number)
+			}
+			Some(RequestBlockId::Tag(RequestBlockTag::Earliest)) => Ok(0),
+			Some(RequestBlockId::Tag(RequestBlockTag::Pending)) => {
+				Err(internal_err("'pending' is not supported"))
+			}
+		}
+	}
+
+	fn handle_request(
+		client: &C,
+		backend: &BE,
+		cached_blocks: &mut BTreeMap<H256, CacheBlock>,
+		req: FilterRequest,
+		touched_blocks: &mut Vec<H256>,
+	) -> Result<Vec<TransactionTrace>> {
+		let from_block = Self::block_id(client, req.from_block)?;
+		let to_block = Self::block_id(client, req.to_block)?;
+
+		if to_block < from_block {
+			return Err(internal_err(format!(
+				"fromBlock ({}) must be greater or equal than ({})",
+				from_block, to_block
+			)));
+		}
+
+		// if from_block == 0 {
+		// 	return Err(internal_err("Tracing genesis block is not possible"));
+		// }
```

Comment:
Commented code. The (non) handling of the genesis block should go in a comment or docs somewhere.